azurerm/internal/services/iothub/iothub_resource.go

```diff
@@ -403,6 +403,11 @@ func resourceArmIotHub() *schema.Resource {
 				},
 			},
 
+			"public_network_access": {
+				Type:     schema.TypeBool,
+				Optional: true,
```

Comment:
does this need to default to enabled? it'd be good to confirm with the portal's behaviour

Reply:
If you don't specify it when creating an iothub, it doesn't get returned by the api so I don't think we can default it without breaking people.

Reply:
You're required to specify it when creating it in the portal. I believe this functionality was added recently. I imagine it's on by default but the api doesn't return it if a user didn't specify it when creating. 

![image](https://user-images.githubusercontent.com/6515518/94014221-189f4400-fd60-11ea-9c76-7ae2a6daccb4.png)